### ISSUE TITLE
`functools.partial` discards signature of dataclass-generated __init__

### ISSUE THREAD

**Issue by jaskp** [microsoft/pyright#10191]
**Describe the bug**
After applying `partial` on a dataclass "C" with an auto-generated `__init__` method, the return type becomes `(...) -> C`

**Code or Screenshots**
```python
import functools

class C:
    def __init__(self, a: int, b: int): ...

functools.partial(C, a=1)() # as expected ✅ Argument missing for parameter "b"
```

```python
import functools, dataclasses

@dataclasses.dataclass(init=False)
class C:
    a: int
    b: int
    def __init__(self, a: int, b: int): ...

functools.partial(C, a=1)() # as expected ✅ Argument missing for parameter "b"
```

```python
import functools, dataclasses

@dataclasses.dataclass
class C:
    a: int
    b: int

partial(C, a=1)() # false negative ❌ No error
```

**VS Code extension or command-line**
Cursor Pyright

**Comment by luwqz1** [comment]
I cannot reproduce your problem, probably you may have an older version of pyright installed

[Pyright is working as intended here](https://pyright-play.net/?pyrightVersion=1.1.398&pythonVersion=3.13&strict=true&code=JYWwDg9gTgLgBAMwK4DsDGMIQDYGcA0cAJgIYwlrYm64CmuAsAFDMACp5l1duAdBxSo1mXGnADCALmZxZcEpLjAUMGXIBGi5apZNk6TDj5gSsYCWwAKcYRIBeAIwBKS0%2BZA)
